src/pkg/blob/manager.go

```diff
@@ -95,22 +99,36 @@ func (m *manager) CleanupAssociationsForArtifact(ctx context.Context, artifactDi
 	return m.dao.DeleteArtifactAndBlobByArtifact(ctx, artifactDigest)
 }
 
-func (m *manager) CleanupAssociationsForProject(ctx context.Context, projectID int64, blobs []*Blob) error {
-	if len(blobs) == 0 {
-		return nil
+func min(a, b int) int {
+	if a < b {
+		return a
 	}
+	return b
+}
 
-	shouldUnassociatedBlobs, err := m.dao.FindBlobsShouldUnassociatedWithProject(ctx, projectID, blobs)
-	if err != nil {
-		return err
+func (m *manager) CleanupAssociationsForProject(ctx context.Context, projectID int64, batches []*Blob) error {
+	if len(batches) == 0 {
+		return nil
 	}
-
-	var blobIDs []int64
-	for _, blob := range shouldUnassociatedBlobs {
-		blobIDs = append(blobIDs, blob.ID)
+	for cursor := 0; cursor < len(batches); cursor += deleteBatchSize {
+
+		// create a batch of size deleteBatchSize or the remaining elements
+		blobs := batches[cursor:min(len(batches)-cursor, deleteBatchSize)]
```

Comment:
I think here should be `blobs := batches[cursor:min(cursor + deleteBatchSize, len(batches))]`

Reply:
Updated